bin/hipcc

```diff
@@ -58,7 +60,18 @@ $isWindows = $^O eq 'MSWin32';
 $HIPCC_COMPILE_FLAGS_APPEND=$ENV{'HIPCC_COMPILE_FLAGS_APPEND'};
 $HIPCC_LINK_FLAGS_APPEND=$ENV{'HIPCC_LINK_FLAGS_APPEND'};
 
-$HIP_PATH=$ENV{'HIP_PATH'} // dirname (dirname $0); # use parent directory of hipcc
+#
+# TODO: Fix rpath LDFLAGS settings
+#
+# Since this hipcc script gets installed at two uneven hierarchical levels,
+# linked by symlink, the absolute path of this script should be used to
+# derive HIP_PATH, as dirname $0 could be /opt/rocm/bin or /opt/rocm/hip/bin
+# depending on how it gets invoked.
+# HIP_PATH points to <rocm_install_dir>/hip. 
+# Defining ROCM_ROOTDIR to be used instead of hardcoded /opt/rocm defaults
+#
+$HIP_PATH=$ENV{'HIP_PATH'} // dirname(Cwd::abs_path("$0/../")); # use parent directory of hipcc
```

Comment:
There is a soft link /opt/rocm/bin/hipcc -> /opt/rocm/hip/bin/hipcc. In many cases hipcc is called through /opt/rocm/bin/hipcc instead of /opt/rocm/hip/bin/hipcc. You need to be able to handle this situation. Probably follow the soft link if hipcc is a soft link?

Reply:
I abs_path would resolve soft link and return the canonical path

Reply:
Yes, @scchan answered your question @yxsamliu 
I have tested it from both places. Please NOTE: the real file MUST exist under /opt/roc/hip/bin/hipcc. You can have symlinks from anywhere to that.